wandb/sdk/artifacts/artifact.py

```diff
@@ -532,6 +533,78 @@ def type(self) -> str:
         """The artifact's type. Common types include `dataset` or `model`."""
         return self._type
 
+    @property
+    @ensure_logged
+    def url(self):
+        """
+        Constructs the URL of the artifact.
+
+        Returns:
+            str: The URL of the artifact.
+        """
+        if self.collection.is_sequence():
+            return self._construct_standard_url()
+        else:
+            if is_artifact_registry_project(self.project):
+                return self._construct_registry_url()
+            elif self._type == "model" or self.project == "model-registry":
+                return self._construct_model_registry_url()
+            else:
+                return self._construct_standard_url()
```

Comment:
That definitely looks better! Re else/elif, yeah to me it looks a bit cleaner.

Tbh I returned "" just because that's what we do for runs (https://github.com/wandb/wandb/blob/059c637d422bf37f355341974faaa18fbd193bea/wandb/sdk/wandb_settings.py#L886), so wanted to be consistent, but happy to raise an error if you think that's better!